adminSiteClient/ChartEditor.ts

```diff
@@ -188,16 +188,19 @@ export class ChartEditor {
         if (json.success) {
             if (isNewGrapher) {
                 this.newChartId = json.chartId
+                this.grapher.id = json.chartId
+                this.savedGrapherJson = JSON.stringify({
+                    ...currentGrapherObject,
+                    id: json.chartId,
+                })
```

Comment:
The purpose of `savedGrapherJson` I think is to detect when a Grapher has changed on the client-side compared to what is in the database:

https://github.com/owid/owid-grapher/blob/e9d60ac5754fac84ad52b06d68866e27b769d03e/adminSiteClient/ChartEditor.ts#L104-L106

Which I think would mean that the order of properties is important? Which means that maybe this should be:

```ts
JSON.stringify(this.grapher.object)
```

Reply:
Good point! The `savedGrapherJson` check doesn't currently work though; `savedGrapherJson` is always initialized to `{}`.